### PR TITLE
[RUN] Implement Gambit, various small fixes

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Note: there is a currently unused function for Rayke in this PR which will be used as soon as Rayke is able to be implemented. I can comment it or remove it out if necessary for style check happiness, but I wanted to "hard" document it so I don't lose the captured data.

### CODE REVIEW

**Comment by zach2good** [comment]
You can appease CI with `utils.unused(...)`

File: scripts/globals/job_utils/rune_fencer.lua [hunk 1]
```diff
@@ -228,6 +245,48 @@ local function getAnimationSwipeLunge(weaponSkillType) -- verified via retail ac
...
+        [xi.skill.GREAT_AXE]    = 29, -- gaxe/scythe share the animation for Rayke
+        [xi.skill.SCYTHE]       = 29,
+        [xi.skill.POLEARM]      = 31,
+        [xi.skill.KATANA]       = 32,
+        [xi.skill.GREAT_KATANA] = 33,
+        [xi.skill.CLUB]         = 28,
+        [xi.skill.STAFF]        = 34,
+    }
+    return weaponAnimationMap[weaponSkillType]
+end
```
Comment (zach2good) [inline scripts/globals/job_utils/rune_fencer.lua:246]: I notice that these three maps are all the same, with increments of 10 added to them, so you could define this as a single table outside the scope of these functions, and then have the functions return `table[weaponSkillType + specificOffset]`

**Review by zach2good — APPROVED** [review]

File: scripts/globals/job_utils/rune_fencer.lua [hunk 2]
```diff
@@ -654,3 +683,47 @@ xi.job_utils.rune_fencer.usePflug = function(player, target, ability, action)
...
+
+    action:speceffect(target:getID(), getSpecEffectElementEffusion(highestRune)) -- set element color for animation.
+    action:setAnimation(target:getID(), getAnimationEffusion(weaponSkillType, 10)) -- set animation for currently equipped weapon
+
+    SDTPower = SDTPower * 100 -- adjust to SDT modifier
+
+    local SDTTypes = {} -- one SDT type per rune which can be additive
+    local i = 0
+
+    for _, effect in ipairs(effects) do
```
Comment (zach2good) [inline scripts/globals/job_utils/rune_fencer.lua]: Do we not have a binding for getrunes? or similar?

Comment (WinterSolstice8) [reply scripts/globals/job_utils/rune_fencer.lua]: Not yet, but I can add one. I'll do that when I get home.

File: src/map/lua/lua_baseentity.cpp [hunk 3]
```diff
@@ -11954,6 +11954,26 @@ void CLuaBaseEntity::reduceBurden(float percentReduction, sol::object const& int
...
+auto CLuaBaseEntity::getAllRuneEffects() -> sol::table
+{
+    auto* PEntity = static_cast<CBattleEntity*>(m_PBaseEntity);
+    std::vector<EFFECT> runeEffectList = PEntity->StatusEffectContainer->GetAllRuneEffects();
+
+    auto table = luautils::lua.create_table();
+
+    for(const auto& runeEffect: runeEffectList) {
+        table.add(runeEffect);
+    }
```
Comment (zach2good) [inline src/map/lua/lua_baseentity.cpp]:
```cpp
for (auto const& runeEffect : runeEffectList)
{
    table.add(runeEffect);
}
```

**Review by zach2good — APPROVED** [review]
All good, just some styling

**Review by claywar — APPROVED** [review]

**Pull request merged**